### DISCUSSION TITLE
Increased memory usage of a solver on current master branch

### DISCUSSION THREAD

**Discussion by gregy4** [google/or-tools#2692] [CP-SAT questions]
Hello,
  I did some performance comparison between 8.2, 9.0 and master branch versions. I found that current version uses more memory, iteration through history showed that the change is caused by commit 08f4a954580693d62ef39be1486bf2c925373f99 commited in Jun 29 and named 'more scheduling propagation/improvements'.

  The change was most visible on big proto file, [used_memory.pdf](https://github.com/google/or-tools/files/6876511/used_memory.pdf). The proto file was executed for 10000s with interleave search and one search worker, current master used nearly 6gb more than version before the change.

  Following proto file was used [test-individual-shifts.zip](https://github.com/google/or-tools/files/6876540/test-individual-shifts.zip).

  Is usage of more memory expected effect of the change or is it an error ?  6gb change on interleave search seems like a lot for me, without interleave search and more search worker the effect should be probably higher.

Jan

**Comment by lperron** [comment]
small comment, interleaved search creates a lot of workers. It is just that they are executed in an interleaved way, instead of in parallel. So the 6Gb memory increase in shared across all workers.
